### PR TITLE
feat(rl): conservative offline RL fine-tune for worker efficiency (#509)

### LINKED ISSUE #509: P1 RL Micro: Offline Conservative RL fine-tune to surpass heuristic worker efficiency
## 问题描述

BC policy 只能复制 heuristic 行为，无法超越。需要用 conservative offline RL 在 BC policy 基础上 fine-tune，产生超越 heuristic 的微操策略。

## 依赖
- Phase M3 (BC policy) 产出初始模型
- #502 (行为遥测) 提供 reward-labeled 样本
- #414 (simulator 可执行) 提供训练数据生成

## 实现目标

1. 从仿真器生成 100,000+ reward-labeled creep-samples
2. 用 CQL (Conservative Q-Learning) 或 IQL (Implicit Q-Learning) fine-tune
3. 在仿真器中评估 RL policy vs heuristic：work_ticks/total_ticks 提升 > 10%
4. 产出 RL model artifact + evaluation report

## 技术方案

- Base policy: BC model from Phase M3
- Algorithm: CQL (conservative, safer for offline data) 或 IQL (simpler, good for mixed-quality data)
- Reward: work_ticks / total_ticks (主要) + energy_delivered (次要) - idle_penalty
- Framework: PyTorch + d3rlpy (offline RL library)

## 安全约束

- 微操层 RL 只控制 task selection (harvest/transfer/build/repair/upgrade) + target selection
- 不控制 creep movement（movement 仍由 Screeps pathfinder 处理）
- 不控制 spawn/construction/territory 决策（这些属于策略层）
- `liveEffect: false` 直到所有 gate 通过

## Acceptance evidence
- RL policy work_ticks ratio > heuristic 10%+ in 5 个不同仿真器场景
- Shadow eval 显示 RL policy 的 KPI 不劣于 heuristic
- Historical validation 通过（不会导致房间丢失）

## Rollback/stop condition
- 如果 RL policy 导致 spawn collapse 或 controller downgrade，立即回退
- 如果 CQL/IQL 收敛不了，尝试更简单的 evolutionary tuning

## Plan reference
`docs/ops/rl-flywheel-v2.md` Phase M4

### PR DESCRIPTION
**Pull request by lanyusea**
Implements a conservative offline RL fine-tuning pipeline for worker efficiency.

## Changes
- **prod/src/rl/workerEfficiency.ts**: Conservative tabular CQL implementation for worker task optimization
- **prod/test/workerEfficiencyRl.test.ts**: Tests for RL training correctness and baseline comparison
- **scripts/screeps_rl_worker_efficiency_train.js**: Training orchestration script
- **docs/ops/rl-worker-efficiency-finetune.md**: Pipeline documentation
- **.gitignore**: RL data directory exclusion

## Verification
```
npm run typecheck  # PASS
npm test -- --runInBand  # 31 suites, 778 tests PASS
npm run build  # PASS
```

Closes #509

### CODE REVIEW

**Comment by lanyusea** [comment]
Review-fix Codex dispatched for the 3 unresolved review threads:

1. **Codex Connector** (PRRT_H6Rd): use held-out split for evaluation gate
2. **CodeRabbit** (PRRT_H62C): eval on held-out scenarios (major)
3. **CodeRabbit** (PRRT_H62D): heuristic fallback when action key absent (major)

Session: proc_fc9ae30b776a, PID 39468, worktree /root/screeps-worktrees/rl-worker-efficiency-509
After Codex commits, controller will verify, push, and run exact-head QA.
